modules/fedimint-dummy/src/db.rs

```diff
@@ -1,12 +1,43 @@
-use fedimint_api::db::DatabaseKeyPrefixConst;
+use fedimint_api::db::{Database, DatabaseKeyPrefixConst, DatabaseVersion, DatabaseVersionKey};
 use fedimint_api::encoding::{Decodable, Encodable};
+use futures::StreamExt;
 use serde::Serialize;
 use strum_macros::EnumIter;
 
+pub const DATABASE_VERSION: DatabaseVersion = DatabaseVersion(1);
+
+pub async fn migrate_dummy_db_version_0(db: &Database) -> Result<DatabaseVersion, anyhow::Error> {
+    let mut read_dbtx = db.begin_transaction().await;
+    let mut write_dbtx = db.begin_transaction().await;
+
+    let example_keys_v1 = read_dbtx
```

Comment:
Now that you've implemented streams here, we could do it via stream to avoid allocating memory for all of them at the same time?

Reply:
Worth checking if it works. :)

Reply:
This is unlikely to work since reading requires a mut ref to the DB tx as well as writing.

Reply:
Yep I did try to use for_each first like this:

`example_keys_v1.for_each(|(key, _)| async {
        let key_v2 = ExampleKey(key.0, "Example String".to_string());
        write_dbtx.insert_new_entry(&key_v2, &()).await.expect("DB Error");
    });`

I tried making the closure `async` and `asycn move` but I'm hitting issues since `write_dbtx` needs to be mut.